### PR TITLE
Upgrade rails to 5.2.0

### PR DESCRIPTION
**Pull request by andrew**
~~Testing out the new release before the final is published, no rush to merge this until then.~~

Updated for rails 5.2.0 final, and unlocks mysql2 and pg gems as well

### CODE REVIEW

**Comment by chrisarcand** [comment]
Nice, can polish and merge this at RailsConf!

**Comment by andrew** [comment]
@chrisarcand sweet! Will be interesting to see if we get any performance improvements from the upgrade: https://oss.skylight.io/app/applications/eLvcBBdLmV6k/recent/6h/endpoints

**Comment by andrew** [comment]
Updated this PR now that 5.2.0 has been published: https://rubygems.org/gems/rails/versions/5.2.0

File: bin/setup [hunk 1]
```diff
@@ -1,10 +1,9 @@
 #!/usr/bin/env ruby
-require 'pathname'
 require 'fileutils'
 include FileUtils
 
 # path to your application root.
-APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
+APP_ROOT = File.expand_path('..', __dir__)
```
Comment (chrisarcand) [inline bin/setup]: I mean ok, but why? Or are you just conforming to what Rails 5 now generates?

File: bin/rails [hunk 2]
```diff
@@ -1,9 +1,4 @@
 #!/usr/bin/env ruby
-begin
-  load File.expand_path('../spring', __FILE__)
-rescue LoadError => e
-  raise unless e.message.include?('spring')
-end
```
Comment (chrisarcand) [inline bin/rails]: ?

File: config/initializers/new_framework_defaults_5_2.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# Use AES-256-GCM authenticated encryption as default cipher for encrypting messages
+# instead of AES-256-CBC, when use_authenticated_message_encryption is set to true.
+# Rails.application.config.active_support.use_authenticated_message_encryption = true
+
+# Add default protection from forgery to ActionController::Base instead of in
+# ApplicationController.
+# Rails.application.config.action_controller.default_protect_from_forgery = true
+
+# Use SHA-1 instead of MD5 to generate non-sensitive digests, such as the ETag header.
+# Rails.application.config.active_support.use_sha1_digests = true
```
Comment (chrisarcand) [inline config/initializers/new_framework_defaults_5_2.rb]: Is this file a new convention or did you just throw the new 5.2 stuff in here yourself?

Comment (andrew) [reply config/initializers/new_framework_defaults_5_2.rb]: I ran `rake app:update` and kept all the changes to the default generated file changes

File: bin/setup (hunk 1 above)
Comment (andrew) [reply bin/setup]: I ran `rake app:update` and kept all the changes to the default generated file changes

**Comment by chrisarcand** [comment]
lol solid responses, thanks @andrew :trollface:

**Comment by jules2689** [comment]
I'd give it a shot on shopify's instance, but we've diverged a bit from master (PRs awaiting review) :(

**Comment by jules2689** [comment]
Ok, actually, I was able to rebase shopify's production branch on our fork with this ... trying it out now

**Comment by jules2689** [comment]
It seems to be working well, :+1:

**Comment by andrew** [comment]
🚢 on https://octobox.io

**Pull request merged**